packages/router/router.ts

```diff
@@ -1074,6 +1077,21 @@ export function createRouter(init: RouterInit): Router {
       return;
     }
 
+    if (
+      to != null &&
+      opts &&
+      "action" in opts &&
+      typeof opts.action === "function"
+    ) {
+      to = null;
+      warning(
+        false,
+        "router.navigate() should not include a `to` location when a custom " +
+          "`action` is passed, the `to` will be ignored in favor of the current " +
+          "location."
+      );
+    }
```

Comment:
@ryanflorence This feels like maybe the API isn't quite right, but I didn't like navigating to a "function"

```ts
// Current API - `to` is ignored when action is passed
// router.navigate(to: To, opts: RouterNavigationOptions)
router.navigate(null, { 
  formMethod: 'post',
  formData: new FormData(),
  action: customAction,
})

// Alternate API - custom action is passed as the `to` value
// router.navigate(to: To | ActionFunction, opts: RouterNavigationOptions)
router.navigate(customAction, {
  formMethod: 'post',
  formData: new FormData(),
});
```